project/Build.scala

```diff
@@ -322,13 +333,37 @@ object Build extends sbt.Build {
       }
   )
 
-  /** Depend library as if (exportJars in library) was set to false */
-  val compileWithLibrarySetting = {
-    internalDependencyClasspath in Compile ++= {
-      val prods = (products in (library, Compile)).value
-      val analysis = (compile in (library, Compile)).value
+  implicit class ProjectOps(val project: Project) extends AnyVal {
+    /** Uses the Scala.js compiler plugin. */
+    def withScalaJSCompiler: Project =
+      if (isGeneratingEclipse) project
+      else project.dependsOn(compiler % "plugin")
```

Comment:
We should probably consolidate this with `myScalaJSSettings` using a `Project` transformer (later, different PR).